iwyu.cc

```diff
@@ -1676,15 +1679,16 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
       return true;
 
     if (HasCovariantReturnType(method_decl)) {
-      const Type* return_type = RemovePointersAndReferencesAsWritten(
-          method_decl->getReturnType().getTypePtr());
+      const Type* return_type = method_decl->getReturnType().getTypePtr();
+      const Type* deref_return_type =
+          RemovePointersAndReferencesAsWritten(return_type);
 
       VERRS(6) << "Found covariant return type in "
                << method_decl->getQualifiedNameAsString()
-               << ", needs complete type of " << PrintableType(return_type)
-               << "\n";
+               << ", needs complete type of "
+               << PrintableType(deref_return_type) << "\n";
```

Comment:
We could also remove the type name, since it will be reported by `ReportTypeUse` at loglevel 3, right?

Something like `Found covariant return type in ..., needs complete type\n"`

Reply:
> since it will be reported by ReportTypeUse at loglevel 3, right

Only when the return type is not defined in the same file and not properly `#include`d, but I agree that such a verbose diagnostic doesn't seem very useful.